components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2450,6 +2492,196 @@ public void updateGroup(Group oldGroup, Group newGroup) throws CharonException {
+        } catch (BadRequestException e) {
+            if (log.isDebugEnabled()) {
+                log.debug(e);
+            }
+            throw new CharonException("Error in updating the group", e);
+        }
+
+        return getGroup(groupId, requiredAttributes);
+    }
+
+    private void prepareAddedRemovedMemberLists(Set<String> addedMembers, Set<String> removedMembers,
+                                                Set<Object> newlyAddedMemberIds, PatchOperation memberOperation,
+                                                Map<String, String> memberObject) throws UserStoreException {
+
+        if (StringUtils.isEmpty(memberObject.get(SCIMConstants.GroupSchemaConstants.DISPLAY))) {
+            List<org.wso2.carbon.user.core.common.User> userListWithID =
+                    carbonUM.getUserListWithID(SCIMConstants.CommonSchemaConstants.ID_URI,
+                            memberObject.get(SCIMConstants.GroupSchemaConstants.VALUE), null);
+            if (isNotEmpty(userListWithID)) {
+                memberObject.put(SCIMConstants.GroupSchemaConstants.DISPLAY, userListWithID.get(0).getUsername());
+                memberOperation.setValues(memberObject);
+            }
+        }
+
+        if (StringUtils.equals(memberOperation.getOperation(), SCIMConstants.OperationalConstants.ADD)) {
+            removedMembers.remove(memberObject.get(SCIMConstants.GroupSchemaConstants.DISPLAY));
+            addedMembers.add(memberObject.get(SCIMConstants.GroupSchemaConstants.DISPLAY));
+            newlyAddedMemberIds.add(memberObject.get(SCIMConstants.GroupSchemaConstants.VALUE));
+        } else if (StringUtils.equals(memberOperation.getOperation(),
+                SCIMConstants.OperationalConstants.REMOVE)) {
+            addedMembers.remove(memberObject.get(SCIMConstants.GroupSchemaConstants.DISPLAY));
+            removedMembers.add(memberObject.get(SCIMConstants.GroupSchemaConstants.DISPLAY));
+        }
+    }
+
+    private void setGroupDisplayName(String oldGroupName, String newGroupName)
+            throws IdentityApplicationManagementException, CharonException, BadRequestException, IdentitySCIMException,
+            UserStoreException {
+
+        String userStoreDomainFromSP = getUserStoreDomainFromSP();
+
+        String oldGroupDomain = IdentityUtil.extractDomainFromName(oldGroupName);
+        if (userStoreDomainFromSP != null && !userStoreDomainFromSP.equalsIgnoreCase(oldGroupDomain)) {
+            throw new CharonException("Group :" + oldGroupName + "is not belong to user store " +
+                    userStoreDomainFromSP + "Hence group updating fail");
+        }
+
+        // If the updated group name does not contain a user store domain, it will be returned as PRIMARY.
+        String updatedGroupDomain = IdentityUtil.extractDomainFromName(newGroupName);
+
+        if (isPrimaryDomain(updatedGroupDomain) && !isPrimaryDomain(oldGroupDomain)) {
+             /*
```

Comment:
Shall we fix alignment issues in here?

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/280/commits/36add81b74fe1bfc479adf2b8217e00af089b8f7